### PR TITLE
layout: Add *very* basic support for table layout

### PR DESCRIPTION
**Pull request by mrobinson**
This is the first step to proper table layout. It implements a naive
layout algorithm, notably only taking into account the preferred widths
of the first table row. Still, it causes some float tests to start
passing, so turn on the `layout.tables.enabled` preference for those
directories.

Co-authored-by: Oriol Brufau <obrufau@igalia.com>

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

### CODE REVIEW

File: tests/wpt/tests/tools/wptrunner/wptrunner/executors/executorservo.py [hunk 1]
```diff
@@ -27,53 +27,24 @@
...
-    return debug_args + command
-
-
-
-class ServoTestharnessExecutor(ProcessTestExecutor):
-    convert_result = testharness_result_converter
-
-    def __init__(self, logger, browser, server_config, timeout_multiplier=1, debug_info=None,
-                 pause_after_test=False, **kwargs):
+class ServoExecutor(ProcessTestExecutor):
```
Comment (Loirooriol) [inline tests/wpt/tests/tools/wptrunner/wptrunner/executors/executorservo.py:30]: Should be rebased now that #31115 has landed

File: tests/wpt/meta/css/CSS2/floats-clear/__dir__.ini [hunk 2]
```diff
@@ -0,0 +1 @@
+prefs: ["layout.tables.enabled:true"]
```
Comment (Loirooriol) [inline tests/wpt/meta/css/CSS2/floats-clear/__dir__.ini:1]: Should probably enable the pref too in `css/CSS2/tables/` and `css/css-tables/`. But the latter are currently skipped, so may need to be unskipped first in another PR, to see the difference here?

File: components/layout_2020/formatting_contexts.rs [hunk 3]
```diff
@@ -206,14 +206,20 @@ impl NonReplacedFormattingContext {
                 fc.layout(layout_context, positioning_context, containing_block)
             },
             NonReplacedFormattingContextContents::Table(table) => {
-                table.layout(layout_context, positioning_context, containing_block)
+                let inline_content_sizes = self.content_sizes.as_ref().cloned();
```
[the remark this answers is not in the document]
Comment (mrobinson) [reply components/layout_2020/formatting_contexts.rs:209]: I think this parameter can be removed entirely because we aren't using it any longer.

File: components/layout_2020/table/mod.rs [hunk 4]
```diff
@@ -2,46 +2,80 @@
...
-        ContentSizes::zero()
+    pub(crate) fn new(style: Arc<ComputedValues>) -> Self {
+        Self {
+            style,
+            slots: Vec::new(),
+            size: TableSize::zero(),
+        }
+    }
+
+    /// Convenience method for get() that returns a SlotAndLocation
```
[the remark this answers is not in the document]
Comment (mrobinson) [reply components/layout_2020/table/mod.rs:51]: I've updated the documentation for this function.

File: resources/prefs.json [hunk 5]
```diff
@@ -100,7 +100,7 @@
   "layout.columns.enabled": false,
   "layout.flexbox.enabled": false,
   "layout.legacy_layout": false,
-  "layout.tables.enabled": false,
```
[the remark this answers is not in the document]
Comment (mrobinson) [reply resources/prefs.json]: Whoops. I've disabled this. It was accidentally left on from our experiments. I think it was useful though, because we know more-or-less how the tests looks now.

**Review by Loirooriol — APPROVED** [review]
